File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dict = {}
    
    for c in clothes:
        if c[1] not in dict:
            dict[c[1]] = 1
        else:    
            dict[c[1]] += 1
    
    for i in dict.values():
        answer *= (i+1)
    
    
    
    return answer -1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> hm = new HashMap<>();
        
        for(String[] c : clothes){
            if(!hm.containsKey(c[1])){
                hm.put(c[1], 1);
            }
            else{
                hm.put(c[1], hm.get(c[1]) + 1);
            }
        }
        
        for(String s : hm.keySet()){
            answer *= (hm.get(s)+1);
        }
        
        
        return answer - 1;
    }
}